### DISCUSSION TITLE
Maximum capacity and performance of tables

### DISCUSSION THREAD

**Discussion by PanYangyi** [jtablesaw/tablesaw#1215] [General]
I am using tablesaw as the engine for table calculation, so I would like to inquire about the maximum table capacity that a single machine can bear? How is the performance? Do you have specific statistical data?

**Comment by lwhite1** [comment]
That's really too general a question to provide a useful answer. It would be best to try it out yourself on the machine you want to use, doing the kinds of work you want it to do, and considering what degree of performance is acceptable to your application. Performance is very different depending on whether you're  sorting, doing aggregations, joining large tables, etc. 

Unfortunately, there is no test suite to run to evaluate these things.  The hard limit is that no column can have more than Integer.MAX_INT values.

**Reply by PanYangyi** [reply]
Thank you for your reply. It is a very good table calculation tool！
